tests/unit/lms/resources/_js_config_test.py

```diff
@@ -0,0 +1,161 @@
+
+
+class TestJSConfigAuthToken:
+    """Unit tests for the "authToken" sub-dict of JSConfig."""
+
+    def test_it(
+        self, authToken, bearer_token_schema, BearerTokenSchema, pyramid_request
+    ):
+        BearerTokenSchema.assert_called_once_with(pyramid_request)
+        bearer_token_schema.authorization_param.assert_called_once_with(
+            pyramid_request.lti_user
+        )
+        assert authToken == bearer_token_schema.authorization_param.return_value
+
+    def test_it_is_None_for_non_lti_users(
+        self, authToken, BearerTokenSchema, context, pyramid_request
+    ):
+        pyramid_request.lti_user = None
+
+        assert JSConfig(context, pyramid_request).config["authToken"] is None
+
+    @pytest.fixture
+    def authToken(self, config):
+        return config["authToken"]
+
+
+class TestJSConfigHypothesisClient:
+    """Unit tests for the "hypothesisClient" sub-dict of JSConfig."""
+
+    def test_it_contains_one_service_config(self, config):
+        assert len(config["services"]) == 1
+
+    def test_it_includes_the_api_url(self, config):
+        assert config["services"][0]["apiUrl"] == "https://example.com/api/"
+
+    def test_it_includes_the_authority(self, config):
+        assert config["services"][0]["authority"] == "TEST_AUTHORITY"
+
+    def test_it_disables_share_links(self, config):
+        assert config["services"][0]["enableShareLinks"] is False
+
+    def test_it_includes_grant_token(self, config):
+        before = int(datetime.datetime.now().timestamp())
+
+        grant_token = config["services"][0]["grantToken"]
+
+        claims = jwt.decode(
+            grant_token,
+            algorithms=["HS256"],
+            key="TEST_JWT_CLIENT_SECRET",
+            audience="example.com",
+        )
+        after = int(datetime.datetime.now().timestamp())
+        assert claims["iss"] == "TEST_JWT_CLIENT_ID"
+        assert claims["sub"] == "acct:example_username@TEST_AUTHORITY"
+        assert before <= claims["nbf"] <= after
+        assert claims["exp"] > before
+
+    def test_it_includes_the_group(self, pyramid_request, config):
+        groups = config["services"][0]["groups"]
+
+        assert groups == ["example_groupid"]
+
+    def test_it_is_empty_if_provisioning_feature_is_disabled(
+        self, context, pyramid_request
+    ):
+        context.provisioning_enabled = False
+
+        assert JSConfig(context, pyramid_request).config["hypothesisClient"] == {}
+
+    def test_it_is_mutable(self, config):
+        config.update({"a_key": "a_value"})
+
+        assert config["a_key"] == "a_value"
+
+    @pytest.fixture
+    def config(self, config):
+        return config["hypothesisClient"]
+
+
+class TestJSConfigRPCServer:
+    """Unit tests for the "rpcServer" sub-dict of JSConfig."""
+
+    def test_it(self, config):
+        assert config == {"allowedOrigins": ["http://localhost:5000"]}
+
+    @pytest.fixture
+    def config(self, config):
+        return config["rpcServer"]
```

Comment:
I'm not sure this fixture makes sense to me though. I think this is only included here because it was useful in the class above. I think this might be premature optimisation.

Reply:
It might as well be consistent with the rest of the test classes in the file, and if we ever send more settings to the rpcServer and need to add more tests to this class we'll be starting off how we mean to continue